tools/hyperref-check.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+# Extracting all broken hyperref detected by PDFTex
+for file in "./main/acle.md" "./morello/morello.md" "./mve_intrinsics/mve.md" "./neon_intrinsics/advsimd.md"; do
+  echo "Checking $file..."
+
+  # The following string of pipes extracts the unresolved internal links by:
+  # - Using the error and warning messages generated by pandoc pdf conversion as the erroutput variable value
+  # Note: A Variable is used instead of a pipe in order to avoid EOF errors occuring due to the -zoP flag.
+  # - Using grep to extract all unresolved links, eg "pdfTeX warning (dest): name{ssec-bf16-scalar}"
+  # Note: Some broken links may be on mutiple lines. The -zoP flag is used to accomodate this. More info here:
+  # https://unix.stackexchange.com/a/361707
+  # - Using the first tr to remove any line breaks in the middle of warning messages
+  # - Using sed to extract only the section name without the warning message eg "ssec-bf16-scalar"
+  # - Using to sort, uniq and tr to sort the links in alphabetical order, only
+  # output each link once and only one per line respectively
+  erroutput=`pandoc $file --verbose -o pdfs/tmp.pdf 2>&1`
+  broken_refs=`grep -zoP 'pdfTeX warning \(dest\): name[\n]?\{[\w|-]*[\n]?[\w|-]*\}' <<< "$erroutput" | \
+               tr -d "\n" | sed -E  's/pdfTeX warning \(dest\): name\{/ /g' | \
+               tr "} " "\n" | sort | uniq`
```

Comment:
This is very nice, congratulations for coming up with this machinery!

However, I think it is somehow fragile, to the point it feels like it needs its own set of unit test to test all the cases you are covering.

Unit testing this script would be too much, so I was thinking whether we can simplify the procedure a bit more.

In my experiments I didn't notice all the corner cases you were trying to cover, therefore the extract-sort-uniq sequence seemed a good idea to get some actionable information out of the script. Especially the fact that the broken links are unique it was good to process them orderly, and fix them one by one.

However, now that they are fixed it is unlikely that they will show up again in huge numbers, but just one or maybe two.

We could then process the content of `erroutput` to extract all the text between the first and the last of the lines in the format of 

```
pdfTeX warning (dest): name{sec-NEON-intrinsics} has been referenced but does n
ot exist, replaced by a fixed one
```

By matching just `pdfTeX warning (dest): name{`.

When we get matches, we print all of them and we prompt the user to run `pandoc $file --verbose -o pdfs/tmp.pdf` to get more actionable infos.

What do you think?

Reply:
I think that's sensible. It shouldn't disrupt anything and doesn't waste time running the script if no errors are found.
When you say the user is prompted, do you mean the CI will require user input? Would that even work?